contracts/button/errors.rs

```diff
@@ -0,0 +1,87 @@
+use ink_env::Error as InkEnvError;
+use ink_prelude::{format, string::String};
+use openbrush::contracts::psp22::PSP22Error;
+
+/// GameError types
+#[derive(Debug, PartialEq, Eq, scale::Encode, scale::Decode)]
+#[cfg_attr(feature = "std", derive(scale_info::TypeInfo))]
+pub enum GameError {
+    /// Returned if reset is called before the deadline
+    BeforeDeadline,
+    /// Returned if button is pressed after the deadline
+    AfterDeadline,
+    /// Returned if a call is made from an account with missing access control privileges
+    MissingRole(String),
+    /// Returned if a call to another contract has failed
+    ContractCall(String),
+}
+
+impl From<PSP22Error> for GameError {
```

Comment:
here's what I'm thinking:
PSP22Error is Decode/Encode anyway, so this whole long cast impl can become just:
```
    impl From<PSP22Error> for GameError {
        fn from(inner: PSP22Error) -> Self {
            GameError::PSP22Error(inner)
        }
    }
```

better yet, since it will be re-used across many contracts, we could provide a marker trait (say `IsError`) and a blanket implementation for all enums that are `IsError`. Then implementing contracts get it for free by `impl IsError for MyContractError`

Reply:
for now we don't have a right place for such common code so I wouldn't introduce it now; however, I simplified the implementation as suggested

Reply:
I have some blackout, but I do not see any gain, since any struct `MyContractError` would have to provide a way to be converted from `PSP22Error` anyway, but this time within the `IsError` trait, right?

and since we postponed extracting common crate with utils (https://cardinal-cryptography.atlassian.net/browse/A0-1380) we might postpone such future-oriented things as well, so that this PR stays as 'squashing three crates into one' rather than 'squashing and pulling out some things that should be useful in other PRs'

Reply:
not exactly like that, what I meant is leveraging [this pattern](https://www.educative.io/answers/definition-blanket-implementation), with IsError being an empty (no associated methods) "marker" trait. You'd implement `From`'s just once for T: Decode + Encode + IsError and as long as your other error enum has these traits it gets the `From` "for free".

As long as it doesn't get lost we can implement in other PR, resolve at will

Reply:
resolving for now